tests/unit/test_vtk.py

```diff
@@ -1,9112 +1,390 @@
+    )
+
+    # If the difference is empty, the meshio objects are identified as identical.
+    return diff == {}
+
+
+# Helper for parametrization of test_single_subdomains. Define collection of
+# single subdomains incl. 1d, 2d, 3d grids, of simplicial, Cartesian and
+# polytopal element type.
+
+# Construct 2d polytopal grid
+sd_polytop_2d = pp.StructuredTriangleGrid([2] * 2, [1] * 2)
+sd_polytop_2d.compute_geometry()
+pp.coarsening.generate_coarse_grid(sd_polytop_2d, [0, 1, 3, 3, 1, 1, 2, 2])
+
+# Construct 3d polytopal grid
+sd_polytop_3d = pp.CartGrid([3, 2, 3], [1] * 3)
+sd_polytop_3d.compute_geometry()
+pp.coarsening.generate_coarse_grid(
+    sd_polytop_3d, [0, 0, 1, 0, 1, 1, 0, 2, 2, 3, 2, 2, 4, 4, 4, 4, 4, 4]
+)
+
+# Define data type for a single subdomain
+SingleSubdomain = namedtuple("SingleSubdomain", ["grid", "ref_vtu_file"])
+
+# Define the collection of subdomains
+single_subdomains = [
+    # 1d grid
+    SingleSubdomain(
+        pp.CartGrid(3, 1),
+        f"{folder_reference}/single_subdomain_1d.vtu",
+    ),
+    # 2d simplex grid
+    SingleSubdomain(
+        pp.StructuredTriangleGrid([3] * 2, [1] * 2),
+        f"{folder_reference}/single_subdomain_2d_simplex_grid.vtu",
+    ),
+    # 2d Cartesian grid
+    SingleSubdomain(
+        pp.CartGrid([4] * 2, [1] * 2),
+        f"{folder_reference}/single_subdomain_2d_cart_grid.vtu",
+    ),
+    # 2d polytopal grid
+    SingleSubdomain(
+        sd_polytop_2d,
+        f"{folder_reference}/single_subdomain_2d_polytop_grid.vtu",
+    ),
+    # 3d simplex grid
+    SingleSubdomain(
+        pp.StructuredTetrahedralGrid([3] * 3, [1] * 3),
+        f"{folder_reference}/single_subdomain_3d_simplex_grid.vtu",
+    ),
+    # 3d Cartesian grid
+    SingleSubdomain(
+        pp.CartGrid([4] * 3, [1] * 3),
+        f"{folder_reference}/single_subdomain_3d_cart_grid.vtu",
+    ),
+    # 3d polytopal grid
+    SingleSubdomain(
+        sd_polytop_3d,
+        f"{folder_reference}/single_subdomain_3d_polytop_grid.vtu",
+    ),
+]
+
+
+@pytest.mark.parametrize("subdomain", single_subdomains)
+def test_single_subdomains(setup, subdomain):
+    # Test of the Exporter for single subdomains of different dimensionality,
+    # and different grid type. Exporting of scalar and vectorial data is tested.
+
+    # Define grid
+    sd = subdomain.grid
+    sd.compute_geometry()
+
+    # Define data
+    dummy_scalar = np.ones(sd.num_cells) * sd.dim
+    dummy_vector = np.ones((3, sd.num_cells)) * sd.dim
+
+    # Export data
+    save = pp.Exporter(
+        sd,
+        setup.file_name,
+        setup.folder,
+        export_constants_separately=False,
+    )
+    save.write_vtu([("dummy_scalar", dummy_scalar), ("dummy_vector", dummy_vector)])
+
+    # Check that exported vtu file and reference file are the same
+    assert _compare_vtu_files(
+        f"{setup.folder}/{setup.file_name}_{sd.dim}.vtu",
+        f"{subdomain.ref_vtu_file}",
+    )
+
+
+def test_mdg_2(setup):
```

Comment:
Maybe change name to test_mdg_2d or something else. The "2" is currently ambiguous

Reply:
Renamed both test_mdg_2 (dropped the 2). And test_mdg_3 (tried to give a more descriptive name).